### PR TITLE
[AArch64] Update feature dep. for Armv9.6 extensions

### PR DESCRIPTION
**Pull request by Lukacma**

### CODE REVIEW

File: llvm/unittests/TargetParser/TargetParserTest.cpp [hunk 1]
```diff
@@ -1861,7 +1881,7 @@ AArch64ExtensionDependenciesBaseArchTestParams
         {AArch64::ARMV8A, {"sme-fa64", "nosme"}, {}, {"sme", "sme-fa64"}},
 
         // sme2 -> {sme2p1, ssve-fp8fma, ssve-fp8dot2, ssve-fp8dot4, sme-f8f16,
-        // sme-f8f32, sme-b16b16}
+        // sme-f8f32, sme-b16b16i, ssve-aes}
```
Comment (jthackray) [inline llvm/unittests/TargetParser/TargetParserTest.cpp:1884]: typo?
```suggestion
        // sme-f8f32, sme-b16b16, ssve-aes}
```

File: llvm/unittests/TargetParser/TargetParserTest.cpp [hunk 2]
```diff
@@ -1926,6 +1952,16 @@ AArch64ExtensionDependenciesBaseArchTestParams
         // rcpc -> rcpc3
         {AArch64::ARMV8A, {"norcpc", "rcpc3"}, {"rcpc", "rcpc3"}, {}},
         {AArch64::ARMV8A, {"rcpc3", "norcpc"}, {}, {"rcpc", "rcpc3"}},
+
+        // sve-aes -> ssve-aes
```
Comment (SpencerAbson) [inline llvm/unittests/TargetParser/TargetParserTest.cpp:1956]:
```suggestion
        // sve2-aes -> ssve-aes
```

File: llvm/unittests/TargetParser/TargetParserTest.cpp [hunk 3]
```diff
@@ -1904,6 +1926,10 @@ AArch64ExtensionDependenciesBaseArchTestParams
         {AArch64::ARMV8A, {"sme-f8f16", "nofp8"}, {}, {"fp8", "sme-f8f16"}},
         {AArch64::ARMV8A, {"nofp8", "sme-f8f32"}, {"fp8", "sme-f8f32"}, {}},
         {AArch64::ARMV8A, {"sme-f8f32", "nofp8"}, {}, {"fp8", "sme-f8f32"}},
+        {AArch64::ARMV9_6A, {"nofp8", "f8f16mm"}, {"fp8", "f8f16mm"}, {}},
```
Comment (SpencerAbson) [inline llvm/unittests/TargetParser/TargetParserTest.cpp:1929]: nit: could you add `f8f16mm` and `f8f32mm` to the `// fp8 -> {...` comment?

**Review by SpencerAbson — APPROVED** [review]
LGTM, thanks.

**Review by jthackray — APPROVED** [review]
Thanks, LGTM

**Pull request merged**